r/inst/build_arrow_static.sh

```diff
@@ -65,6 +65,7 @@ esac
 mkdir -p "${BUILD_DIR}"
 pushd "${BUILD_DIR}"
 ${CMAKE} -DARROW_BOOST_USE_SHARED=OFF \
+    -DARROW_SNAPPY_USE_SHARED=OFF \
```

Comment:
This prevents shared (homebrew) snappy from being used on the builder.  Are there other dependencies we should add here?

Reply:
`ARROW_DEPENDENCY_USE_SHARED` should be the default for all the more specific versions, so just setting that should work?